refract-spec.md

```diff
@@ -107,34 +89,516 @@ In addition to expressing Refract Elements as objects with `element`, `attribute
+### Ref (Element)
+
+The `ref` element MAY be used to reference elements in remote documents or elements in the local document.
+
+1. If the content is a relative fragment identifier, it references an element found in the local document
+1. If there is no fragment identifier, the reference is to the entire document.
+1. If there are multiple elements in a document with the same ID, the first element with the matching value SHOULD be used.
+
+The `ref` element transcludes the contents of the element into the document in which it is referenced.
 
-This is how it would expressed in the compact version.
+#### Properties
+
+- `element` ref (string, fixed)
+- `content` (string) - A URL or fragment identifier. If the content is an absolute URL with a fragment identifier, the fragment identifer references the ID in the corresponding document.
+
+#### Examples
+
+Elements MAY be referenced in remove or local documents.
+
+##### Referencing Remote Element
 
 ```json
-["foo", {}, "bar"]
+{
+  "element": "ref",
+  "content": "http://example.com/document#foo"
+}
+```
+
+##### Referencing Local Elements
+
+```json
+{
+  "element": "ref",
+  "content": "#foo"
+}
+```
+
+## Namespacing
+
+Namespaces provide a way to include element definitions and constraints into another document. The following rules and constraints apply to namespaces.
+
+1. If there is a conflict, the namespaced element defined or referenced last MUST take precedence
+1. Types defined within the local namespace MUST take precedence over referenced namespaces
+1. Prefixed namespaces are only accessible through using a prefix
+1. Namespaces apply to the element in which they are introduced and all child elements
+1. Namespaces MUST not include the elements nor the data in the elements referenced, but rather include the constraints and conditions around the types defined within that namespace
+
+### Namespaces (Element)
+
+The `namespaces` element is a list of Namespace elements.
+
+#### Properties
+
+- `element` namespaces (string, fixed)
+- `content` (array[Namespace])
+
+### Namespace (enum)
+
+#### Members
+
+- (string) - A URL to a namespace
+- (object) - A prefixed namespace
+    - `prefix` (string) - A prefix for the given namespace
+    - `href` (string) - A URL to a namespace
+
+#### Example Using Both String and Object Namespace Elements
+
+This example shows a namespace being included by way of a string, and a prefixed namespace.
+
+```json
+{
+  "element": "foo",
+  "attributes": {
+    "namespaces": [
+      "http://example.com/namespace1",
+      { "prefix": "ns2", "href": "http://example.com/namespace2" }
+    ]
+  }
+}
+```
+
+### Prefix (Element)
+
+A prefix is a way to reference elements in prefixed namespaces.
+
+#### Properties
+
+- `element` prefix (string, fixed)
+- `content` (string) - Prefixed name for given namespace
+
+#### Example
+
+This example says the `foo` element is found in the prefixed namespaced prefixed as `ns2`.
+
+```json
+{
+  "element": "foo",
```

Comment:
@smizell Your recent comment still doesn't address the deprecancy. The `element` described using MSON still says it should be fixed at `prefix` but the example contains `foo` value for `element`.